src/Microsoft.AspNetCore.Dispatcher/DispatcherBase.cs

```diff
@@ -1,13 +1,42 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Dispatcher
 {
-    public abstract class DispatcherBase
+    public abstract class DispatcherBase : IDispatcherOptionsValueProvider
```

Comment:
Convenience base class for dispatchers. It provides you with some services (assuming we add more services) and lets you register it with DI.